MetadataExtractor/Formats/Jpeg/JpegDhtReader.cs

```diff
@@ -42,18 +43,23 @@ public sealed class JpegDhtReader : IJpegSegmentMetadataReader
     {
         ICollection<JpegSegmentType> IJpegSegmentMetadataReader.SegmentTypes => new[] { JpegSegmentType.Dht };
 
-        private HuffmanTablesDirectory directory = null;
-
         public DirectoryList ReadJpegSegments(IEnumerable<JpegSegment> segments)
         {
-
-            var directory = new HuffmanTablesDirectory();
-            foreach (var segment in segments)
+            // This Extract structure is a little different since we only want to return one HuffmanTablesDirectory
+            // for one-to-many segments
+            var directories = new List<Directory>();
+            if (segments.Count() > 0)
```

Comment:
This method ends up enumerating `segments` twice. It'd be good to do it only a single time. Perhaps declare `directory` as `null` initially, then allocate lazily within the `foreach` loop and only add it after the loop if it's non-`null`.